packages/fxa-payments-server/src/lib/test-utils.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useContext, ReactNode } from 'react';
+import { Provider as ReduxProvider } from 'react-redux';
+import { AppContext, AppContextType } from '../../src/lib/AppContext';
+import { config, updateConfig } from '../../src/lib/config';
+import ScreenInfo from '../../src/lib/screen-info';
+import { ReactStripeElements } from 'react-stripe-elements';
+
+import { State } from '../../src/store/types';
+import { createAppStore } from '../../src/store';
+
+declare global {
+  namespace NodeJS {
+    interface Global {
+      fetch: Function;
+    }
+  }
+}
+
+export const wait = (delay: number) =>
+  new Promise(
+    resolve => setTimeout(resolve, delay)
+  );
+
+export const mockConfig = {
+  productRedirectURLs: {
+    product_8675309: 'https://example.com/product',
+  },
+  servers: {
+    content: {
+      url: 'https://content.example',
+    },
+    auth: {
+      url: 'https://auth.example',
+    },
+    oauth: {
+      url: 'https://oauth.example',
+    },
+    profile: {
+      url: 'https://profile.example',
+    },
+  },
+};
+
+export const mockServerUrl = (name: 'auth' | 'oauth' | 'profile' | 'content') =>
+  config.servers[name].url;
+
+export const setupMockConfig = (config?: typeof mockConfig) => {
+  updateConfig(config || mockConfig);
+};
+
+// Minimal mock for react-stripe-elements that lets us trigger onChange
+// handlers with testing data
+const MockStripeElement = ({ testid }: { testid: string }) =>
+  class extends React.Component {
+    _ref: null;
+    setRef: (el: any) => void;
+
+    constructor(props: ReactStripeElements.ElementProps) {
+      super(props);
+
+      // Stash this element's onChange handler in module-global registry.
+      const { onChange } = props;
+      mockStripeElementOnChangeFns[testid] = onChange as onChangeFunctionType;
+
+      // Real react-stripe-elements stash a ref to their container in
+      // this._ref, which we use for tooltip positioning
+      this._ref = null;
+      this.setRef = el => (this._ref = el);
+    }
+
+    render() {
+      return (
+        <div ref={this.setRef} data-testid={testid}>
+          {testid}
+        </div>
+      );
+    }
+  };
+
+// onChange handler registry - indexed by per-component testid, of which
+// there should only be one instance per PaymentForm
+type onChangeFunctionType = (
+  value: stripe.elements.ElementChangeResponse
+) => void;
+
+export const mockStripeElementOnChangeFns: {
+  [name: string]: onChangeFunctionType;
+} = {};
+
+export type MockStripeType = {
+  createToken(
+    options?: stripe.TokenOptions
+  ): Promise<ReactStripeElements.PatchedTokenResponse>;
+};
+
+export type MockStripeContextType = {
+  mockStripe?: MockStripeType;
+};
+
+const MockStripeContext = React.createContext<MockStripeContextType>({});
+
+function injectStripe<P extends Object>(
```

Comment:
Looks complicated, but I kind of just copied & pasted type annotations from the original `injectStripe` until I got this mock working 😓